CMSgov/managed-care-review
# MR-2231: State can add an additional rate cert

## Summary
[MR-2231](https://qmacbis.atlassian.net/browse/MR-2231)

Allow state users to add additional rate certifications.

#### Related issues

#### Screenshots

#### Test cases covered
- Tests for this new feature will be added in another small feature.



## QA guidance
**The new additional rate feature is hidden behind the `multi-rate-submissions` feature flag.**

There have been changes in how the API checks rate data. Please make sure to check
- Changing submission to `contract only` and submitting removes rate data.
   - You can check this by changing it back to contract rates after submission, there should not be any rate data prefilled in the rate details page.
- There has been a change in the domain model since we added `rateInfos`. Since both `makeDocumentKeyLookupList.test.ts` and `makeDocumentDateLookupTable.test.ts` both used  protobuf encoded string, the tests failed because the encoded string did not reflect the changes. I refactored both theses tests to use encoded ( `toProtobuffer()` ) mock revision form data. This way it would reflect changes to both protobuf and domain. 
   - **Please test to make sure documents still show up correctly on the review and summary pages with single rate certifications**
- CSV data dumps will also need QA
